src/lib/core/CHIPTLV.h

```diff
@@ -465,6 +465,25 @@ class DLL_EXPORT TLVReader
      */
     CHIP_ERROR Get(chip::ByteSpan & v);
 
+    /**
+     * Get the value of the current element as an enum value, if it's an integer
+     * value that fits in the enum type.
+     *
+     * @param[out] v Receives the value associated with current TLV element.
+     */
+    template <typename T, typename = std::enable_if_t<std::is_enum<T>::value>>
+    CHIP_ERROR Get(T & v)
+    {
+        typename std::underlying_type_t<T> val;
```

Comment:
I had tried to figure out how to write this before, but couldn't crack it. This was the final thorn in the generated cluster stuff, so thank you for figuring it out!